Source/Scene/Cesium3DTileBatchTableResources.js

```diff
@@ -282,27 +297,116 @@ define([
             result);
     };
 
+    // TODO : put this helper somewhere else?
+    function getComponentCountFromType(type) {
+        switch (type) {
+            case 'SCALAR':
+                return 1;
+            case 'VEC2':
+                return 2;
+            case 'VEC3':
+                return 3;
+            case 'VEC4':
+                return 4;
+            case 'MAT2':
+                return 4;
+            case 'MAT3':
+                return 9;
+            case 'MAT4':
+                return 16;
+            default:
+                throw new DeveloperError('type is not a valid value.');
+        }
+    }
+    
+    // TODO : put this helper somewhere else?
+    function getClassFromType(type) {
+        switch (type) {
+            case 'SCALAR':
+                return undefined;
+            case 'VEC2':
+                return Cartesian2;
+            case 'VEC3':
+                return Cartesian3;
+            case 'VEC4':
+                return Cartesian4;
+            case 'MAT2':
+                return Matrix2;
+            case 'MAT3':
+                return Matrix3;
+            case 'MAT4':
+                return Matrix4;
+            default:
+                throw new DeveloperError('type is not a valid value.');
+        }
+    }
+
+    Cesium3DTileBatchTableResources.prototype.setBatchTable = function(json, binary) {
```

Comment:
Is it reasonable to replace this function with the constructor?  Or does that make the users of this class more difficult to implement?